### ISSUE TITLE
Querying on a SummingMergeTree table that has a projection fails when row policy is used 

### ISSUE THREAD

**Issue by rzilkha** [ClickHouse/ClickHouse#49017]
It seems that on version 22.5, when querying a summingMergeTree that has a projection on top of it, with a row policy fails due to **NOT_FOUND_COLUMN_IN_BLOCK**

### Reproduction steps:
add to config.xml:
    
    <custom_settings_prefixes>cust_</custom_settings_prefixes>
![image](https://user-images.githubusercontent.com/17226062/233665313-af189754-2b79-43c6-a5c9-0150f54ed3b7.png)

and restart clickhouse 

**create database and raw table**

`create database example_db `

```
CREATE TABLE IF NOT EXISTS example_db.raw_example
(
       timestamp UInt32,
    policy_group LowCardinality(String),
    user_id Int64,
    visit UUID

)
    ENGINE = MergeTree
    PARTITION BY (toRelativeWeekNum(toDateTime(timestamp)))
    ORDER BY (policy_group, timestamp);
```

**create target table:**
`CREATE TABLE IF NOT EXISTS example_db.example_target
(
    timestamp UInt32,
    policy_group LowCardinality(String),
    user_id Int64,
    uniq_visits AggregateFunction(uniqCombined,UUID)
    )
    ENGINE = SummingMergeTree()
    PARTITION BY (toRelativeWeekNum(toDateTime(timestamp)))
ORDER BY  (policy_group, timestamp);`

**create mv :**

`CREATE MATERIALIZED VIEW if not exists example_db.example_mv_new
TO example_db.example_target as
select
               timestamp as timestamp,
              policy_group policy_group,
              user_id user_id,
              uniqCombinedState(visit) as uniq_visits
   from example_db.raw_example 
      group by  policy_group, timestamp, user_id`

**insert data to raw table**

`insert into 
      example_db.raw_example 
      select 1679338800, 'policy_1','5',  * from 
      	generateRandom('visit UUID') limit 5`

**create projection**
`ALTER TABLE example_db.example_target ADD PROJECTION if not exists projection_test ( SELECT
            policy_group,
            timestamp,
            user_id,
            uniqCombinedMerge(uniq_visits) AS uniq_visits
        GROUP BY
            policy_group,
            timestamp,
            user_id)`

**apply on past data**

`ALTER TABLE example_db.example_target MATERIALIZE PROJECTION projection_test`

**create user** 

`create user read_only_user`

**grant read access to read_only_user**
` grant select on example_db.* to read_only_user`

**log into clickhouse using that user** 

clickhouse-client -uread_only_user

**run query** 

select  `user_id`,uniqCombinedMerge(`uniq_visits`) as `uniq_visits`  from example_db.example_target  where ( `timestamp` between 1679338800 and 1702013600) group by  `user_id` order by `uniq_visits` desc limit 200 settings cust_policy='policy_1'

you will be able to see the result
![image](https://user-images.githubusercontent.com/17226062/233664282-8c8309a0-6630-44a6-bd6d-5b821db816d8.png)

**go back to user default  , log out and log back to clickhouse-client, now add a row policy** 

`CREATE ROW POLICY IF NOT EXISTS "YYY" ON example_db.example_target FOR SELECT USING policy_group = getSetting('cust_policy') TO read_only_user`

go back to user
clickhouse-client -uread_only_user

**run the query again** 

select user_id,uniqCombinedMerge(uniq_visits) as uniq_visits from example_db.example_target where ( timestamp between 1679338800 and 1702013600) group by user_id order by uniq_visits desc limit 200 settings cust_policy='policy_1'

you will receive 
![image](https://user-images.githubusercontent.com/17226062/233661689-04a26370-d40d-4884-ae45-61adc3b5fd03.png)

now go back to user default and run drop projection

`alter table example_db.example_target drop projection projection_test`

**go back to user read_only_user and see that the query produces results and doesn't fail** 

select user_id,uniqCombinedMerge(uniq_visits) as uniq_visits from example_db.example_target where ( timestamp between 1679338800 and 1702013600) group by user_id order by uniq_visits desc limit 200 settings cust_policy='policy_1'

**Comment by amosbird** [comment]
Can you reproduce the issue in later versions? It's likely already fixed. BTW, what's the actual content of `getSetting('cust_policy')`?

**Comment by rzilkha** [comment]
@amosbird seems that it is fixed on 23.3 , thx.
